### ISSUE TITLE
Make rcheevos and lua as class libraries instead of submodules (Proposal)

### ISSUE THREAD

**Issue by SyrianBallaS** [RetroAchievements/RAIntegration#191]
So while trying to hunt down for code defects and determining where exception handling needs to be targeted, it's starting to get very tedious to suppress warnings for "library" files. The libraries in this case are lua and rcheevos. The reason is that most of the warnings are coming from the source files. We should only be using the header files of libraries anyway. In the meantime, I'll just keep on suppressing warnings. It's not too big a deal, but is something worth considering.

Usually, people using a library are not supposed to make changes to said library. Imagine of someone started changing standard library files or windows sdk files?

Lua should already be taken care of, they have a package on NuGet (Windows) which is at version 5.3.5. I couldn't find a Lua `.so` (Linux/Mac) package on bower or yarn. So people might have to use apt-get or yum, it should install in the system include path for those OS's.

There's only two approaches I can think of to resolve this issue, either suppress the warnings in the library it came from (_might_ be possible with rcheevos but **not** lua) or make them class libraries (`.dll;.so`). They might need library API functionality (export/import).

I can volunteer to make a class library of rcheevos to put on NuGet. I just need to know some requirements.

Which platforms/Operating Systems need to be supported?
Which architecture needs to be supported?

It's very easy to do (even easier in .NET Core) but if anyone wants to learn how to do it for future use, I can leave a tutorial doc in the repo. If it's desired to have precompiled binaries for Linux/Mac, I can do that too.

**Comment by Jamiras** [comment]
I'm not sure I want to go all the way to NuGet, but I think moving them into a separate project seems reasonable. I recommend making a static lib though, as the client DLL needs to be a single download. Also, a static lib should eliminate the need for exports.

You can see some of the other hoops I've had to jump through to make them work as part of the current compilation: https://github.com/RetroAchievements/RAIntegration/blob/master/src/RA_Integration.vcxproj#L38-L68
Basically, overriding project settings for the individual files.

Note that for our particular use case, lua is part of rcheevos, so there would just be the single `rcheevos.lib` (containing rcheevos and lua). And while not directly used by lua or rcheevos, it might make sense to move `md5.c` into the library as well as it's third-party code.

**Comment by SyrianBallaS** [comment]
Seems #193 got accepted. Closing.
